### PR TITLE
React Capstone Project : Music Collection

### PR DESCRIPTION
**Pull request by chudisoft**
In this capstone, I implemented the following as per the project requirements:

The home page.

    When the page loads, the webapp shows the list of music categories and their images.
    When the user clicks (or taps) on a category item, the application navigates to the details page.

Details page.

    In the details page, the webapp retrieves data from the API to show the music of that specific category.
    When the user clicks on the "Back" button (<), the user navigates to the home page.:

Technical requirements

    The project is a single-page application (SPA) built with React and Redux.
    The data retrieved from the API has been stored in the Redux store.
    Filtered the data retrieved from the API using a Filter stateless component.
    Every page (the main page and the pages for each item) has a unique route within the SPA.
    The project is deployed and accessible online using [Render](https://render.com/), [Netlify](https://www.netlify.com/), or [GitHub Pages](https://pages.github.com/).
    Added a link to the online version of the app to the README file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,236 +1,169 @@
...
+- **[.gitignore file]**
 
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
 
 <!-- LIVE DEMO -->
 
 ## 🚀 Live Demo <a name="live-demo"></a>
 
-> Add a link to your deployed project.
+> This Project is deployed at [render.com](https://metrics-webapp-egq7.onrender.com)
```
Comment (topeogunleye) [inline README.md]: - [ ] Great job deploying your project using render, however, please kindly ensure that the deployed project is working.

image

Remember, you're making good progress!. Keep up the good work!

File: src/components/MusicList/MusicList.jsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+import ReactPaginate from "react-paginate";
+import "./MusicList.css";
+import Music from "./Music";
+import { fetchMusicList } from "../../redux/musiclist/musiclistSlice";
+import { Col, Row } from "react-bootstrap";
+import { faSearch } from "@fortawesome/fontawesome-free-solid";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import Loader from "../Loader/Loader";
+
+function MusicList({ category }) {
```
Comment (topeogunleye) [inline src/components/MusicList/MusicList.jsx]: - [ ] Great job implementing the Homepage, however there are a couple of points to consider for further improvement and alignment with the specified requirements:

- I'm not able to view the details page. 
- The home page should have a list of categories that could be filtered using given parameters. please add a filter on the homepage.
- kindly ensure there is a back button on the details page to go back to the homepage.

https://www.loom.com/share/50f5175af0ea42e4a5cda9ae496a38b7?sid=38936eb7-61f4-4721-9465-708ad8d451a7

File: src/components/MusicList/Music.jsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faHeadphones, faPlay } from '@fortawesome/fontawesome-free-solid';
+
+const Music = ({ music }) => {
+  Music.propTypes = {
+    music: PropTypes.object.isRequired,
+  };
+
+  return (
+    <div className="music">
```
Comment (topeogunleye) [inline src/components/MusicList/Music.jsx:10]: Great job on implementing the Music Home page! It looks like you've made progress towards meeting the project requirements for creating the Home page. Your code seems to be on the right track, but there are a couple of points to consider for further improvement and alignment with the specified requirements:

- Kindly follow the [design guidelines](https://www.behance.net/gallery/31579789/Ballhead-App-\(Free-PSDs\)) including:

- **space between elements:** There is no space between items in the Behance design.
- **two color shades:**. The Behance design uses two alternating shades of a color for the items on the home page.



|  Your Design | Behance Design |
| ------------- | ------------- |
| image  | image  |

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @chudisoft, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Suggested changes
## (Highlights) Good Points: 
* Added Home page. ✔️ 
* Added unit tests. ✔️ 
* Added video description. ✔️ 

_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Thanks for submitting your pull request. However, please note that it is important to use descriptive branch names that reflect the changes made in the branch. Using "feature" as a branch name is not descriptive enough and can make it difficult for other developers to understand what changes were made in the branch. Please consider using a more descriptive branch name that reflects the changes made in the branch.
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/index.css [hunk 4]
```diff
@@ -201,9 +201,8 @@ footer {
   .navbar__header {
     padding: 0 2rem;
   }
-  
+
   .container {
     padding: 2em 1em !important;
   }
-
 }
```
Comment (algerina) [inline src/index.css]: Well done working on your project, you've implemented most of the requirements. you still have 2 small isuues to fix:
- [ ] The live version is not working correctly
- [ ] the two color shades:. The Behance design uses two alternating shades of a color for the items on the home page, you are now using 2 color shades but try to alternate them as per Behance design as requested before.


Screenshot 2023-09-09 at 3 57 21 PM

**Review by algerina — CHANGES_REQUESTED** [review]
Hi @chudisoft ,

Great job so far! 

May I highlight the following: 
-  Your app is almost working well ✔
- Well done adding a filter on the homepage✔️



## CHANGES REQUIRED:muscle::clap:


Check the comments under the review.



 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

_Reviewed by [Amel Khiri ](https://github.com/algerina)_

**Review by prolajumokeoni — APPROVED** [review]
Hi @chudisoft 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
 - Deployed link works properly ✅ 
 - Added alternate colors 👏🏿   

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

 - [ ] Your  [branch name](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md#5-basic-rules-to-follow-in-microverse-projects) should always have a name of a feature or action related to your project 👍🏿 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**